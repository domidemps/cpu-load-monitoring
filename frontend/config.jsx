export const CONFIG = {
  pingInterval: 10, // seconds
  eventMinimumDuration: 120, // seconds
  historyWindow: 600, // seconds
}
