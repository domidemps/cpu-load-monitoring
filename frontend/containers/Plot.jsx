/** @jsx jsx */
import {css, jsx} from '@emotion/core'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'

const styles = css`
  margin: 25px;
  padding: 15px;
`

export default function Plot() {
  return (
    <Paper elevation={1} css={styles}>
      <div>
        <Typography
          variant="h4"
          css={css`
            text-align: center;
          `}>
          Average CPU load over time
        </Typography>
      </div>
    </Paper>
  )
}
