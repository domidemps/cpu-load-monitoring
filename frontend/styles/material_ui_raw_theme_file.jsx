import {createMuiTheme} from '@material-ui/core/styles'
import {fade} from '@material-ui/core'

const colors = {
  background: '#1f1f1f',
  paper: '#2f2f2f',
  text: '#e0e0e0',
}

const rawBaseTheme = {
  palette: {
    type: 'dark',
    background: {
      default: colors.background,
      paper: colors.paper,
    },
    text: {
      primary: colors.text,
      secondary: fade(colors.text, 0.7),
    },
  },
  overrides: {
    MuiAppBar: {
      colorPrimary: {
        color: colors.text,
        backgroundColor: colors.paper,
      },
    },
  },
}

export default createMuiTheme(rawBaseTheme)
