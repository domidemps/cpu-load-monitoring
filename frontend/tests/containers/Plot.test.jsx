import React from 'react'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer'

import Plot from '../../containers/Plot'
import {COLORS} from '../../styles/material_ui_raw_theme_file'
import {CONFIG} from '../../config'

describe('Plot', () => {
  let store

  it('renders correctly the CPU load over time', () => {
    store = configureStore([thunk])({
      load: {
        loadOverTime: new Array(CONFIG.historyWindow / CONFIG.pingInterval).fill().map((_, i) => ({
          load: 50,
          time: '12:00:00',
          color: COLORS.normalLoad,
        })),
      },
    })
    expect(
      renderer.create(
        <Provider store={store}>
          <Plot />
        </Provider>,
      ).toJSON(),
    ).toMatchSnapshot()
  })
})
