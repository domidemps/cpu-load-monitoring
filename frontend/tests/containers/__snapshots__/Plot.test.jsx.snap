// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plot renders correctly the CPU load over time 1`] = `
<div
  className="MuiPaper-root css-jfobc3-Plot MuiPaper-elevation5 MuiPaper-rounded"
>
  <div
    style={
      Object {
        "height": 550,
        "width": "100%",
      }
    }
  >
    <h5
      className="MuiTypography-root css-4wzwv1-Plot MuiTypography-h5"
    >
      Average CPU load over time
    </h5>
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
